File: hexample-parent/domain/src/main/java/at/willhaben/sp/hexample/domain/model/JobOffer.java
package at.willhaben.sp.hexample.domain.model;

import java.time.LocalDate;

public class JobOffer {

    private String title;
    private LocalDate publishingDate;

    /**
     * A non-anemic domain model allows you to have certain inner business definitions
     * along the objects
     */
    public Boolean isABrandNewJob() {
        LocalDate yesterday = LocalDate.now().minusDays(1);
        return publishingDate != null && publishingDate.isAfter(yesterday);
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public LocalDate getPublishingDate() {
        return publishingDate;
    }

    public void setPublishingDate(LocalDate publishingDate) {
        this.publishingDate = publishingDate;
    }

}

File: hexample-parent/domain/src/main/kotlin/at/willhaben/sp/hexample/domain/model/JobOffer.kt
package at.willhaben.sp.hexample.domain.model

import java.time.LocalDate

data class JobOffer(
    val title: String,
    val publishingDate: LocalDate? = null
) {
    /**
     * A non-anemic domain model allows you to have certain inner business definitions
     * along the objects
     */
    fun isABrandNewJob(): Boolean {
        val yesterday = LocalDate.now().minusDays(1)
        return publishingDate?.isAfter(yesterday) ?: false
    }
}
